### ISSUE TITLE
sync fails after running out of RAM when using Relay/OCP

### ISSUE THREAD

**Issue by columbian-chris** [10up/ElasticPress#3991]
We utilize Object Caching Pro with a technology called [Relay](https://relay.so) that speeds up Redis object caching by storing a super-efficient partial replica of Redis data in PHP’s memory.  This offers a lot of performance benefits; however, when doing a full Elasticpress sync, I've discovered it uses up the system RAM a whole lot faster.  Even with our 64gb RAM server instance, the sync crashes after the memory usage exceeds 30gb.  In order to complete a full sync, I am forced to turn off Relay/OCP and temporarily switch to just using Redis object caching (without Relay).

Is there any possible way to run a sync without using object caching?

**Comment by felipeelia** [comment]
I don't think there is an easy way to do that @columbian-chris, but perhaps you could use the [`ep_stop_the_insanity`](https://github.com/10up/ElasticPress/blob/develop/includes/classes/IndexHelper.php#L1326) action to reset something?

**Comment by columbian-chris** [comment]
I got some outside help and I'll post my result here in case it helps anyone else.  

I set `vm.swappiness` to `0` in my `/etc/sysctl.conf` file and rebooted, and I reduced my Redis max memory setting.  The result was a successful sync of 571,933 posts with a max peak of 4.3gb.

It was also recommended to me to try editing the PHP-FPM workers setting(s) (pm children) and possibly the `relay.maxmemory` setting in the `relay.ini` module settings file (mine is located in `/etc/php/8.x/mods-available/relay.ini`).
